os/arch/arm/src/amebad/amebad_rtc.c

```diff
@@ -0,0 +1,246 @@
+
+/************************************************************************************
+ * Public Data
+ ************************************************************************************/
+
+/* Variable determines the state of the RTC module.
+ *
+ * After initialization value is set to 'true' if RTC starts successfully.
+ * The value can be changed to false also during operation if RTC for
+ * some reason fails.
+ */
+
+volatile bool g_rtc_enabled;
+
+/************************************************************************************
+ * Private Functions
+ ************************************************************************************/
+
+/************************************************************************************
+ * Public Functions
+ ************************************************************************************/
+
+#if defined(CONFIG_RTC_DATETIME)
+/****************************************************************************
+ * Name: up_rtc_getdatetime
+ *
+ * Description:
+ *   Get the current date and time from the date/time RTC.  This interface
+ *   is only supported by the date/time RTC hardware implementation.
+ *   It is used to replace the system timer.  It is only used by the RTOS
+ *   during initialization to set up the system time when CONFIG_RTC and
+ *   CONFIG_RTC_DATETIME are selected (and CONFIG_RTC_HIRES is not).
+ *
+ *   NOTE: Some date/time RTC hardware is capability of sub-second accuracy.
+ *   That sub-second accuracy is lost in this interface.  However, since the
+ *   system time is reinitialized on each power-up/reset, there will be no
+ *   timing inaccuracy in the long run.
+ *
+ * Input Parameters:
+ *   tp - The location to return the high resolution time value.
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int up_rtc_getdatetime(FAR struct tm *tp)
+{
+	time_t timer;
+	int ret;
+	timer = rtk_rtc_read();
+
+	ret = gmtime_r(&timer, tp);
+	if (ret == 0) return -EOVERFLOW;
+
+	return OK;
+}
+
+#elif defined(CONFIG_RTC_HIRES)
+/****************************************************************************
+ * Name: up_rtc_gettime
+ *
+ * Description:
+ *   Get the current time from the high resolution RTC clock/counter. This
+ *   interface is only supported by the high-resolution RTC/counter hardware
+ *   implementation. It is used to replace the system timer.
+ *
+ * Input Parameters:
+ *   tp - The location to return the high resolution time value.
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int up_rtc_gettime(FAR struct timespec *tp)
+{
+	time_t timer;
+	int ret;
+	timer = rtk_rtc_read();
+
+	ret = gmtime_r(&timer, tp);
+	if (ret == 0) return -EOVERFLOW;
```

Comment:
Pls move return statement to next line and enclose with braces, according to TizenRT coding guidelines.
You may refer to https://github.com/Samsung/TizenRT/wiki/Documentations#coding-rules

Reply:
@kishore-sn We dont not request to apply TizenRT coding rules for chiset vendor code.